### PR TITLE
adds custom dependabot file location support

### PR DESCRIPTION
**Pull request by DaleMckeown**
Should resolve https://github.com/tinglesoftware/dependabot-azure-devops/issues/574

### CODE REVIEW

File: extension/task/utils/parseConfigFile.ts [hunk 1]
```diff
@@ -24,11 +24,15 @@ import axios from "axios";
...
  */
 async function parseConfigFile(variables: ISharedVariables): Promise<IDependabotConfig> {
-  const possibleFilePaths = [
+  let possibleFilePaths = [
     "/.github/dependabot.yaml",
     "/.github/dependabot.yml",
   ];
 
+  if (variables.configFileLocation !== "") {
+    possibleFilePaths = [variables.configFileLocation];
```
Comment (DaleMckeown) [inline extension/task/utils/parseConfigFile.ts:33]: Should this be ``possibleFilePaths = [`"${variables.configFileLocation}"`];`` to maintain quotes around the string? Apologies, not overly familiar with TS.

**Comment by mburumaxwell** [comment]
@DaleMckeown thanks for your contribution. However, locking the configuration file path to be under the .github folder happened some months ago because the loading of the `ignore` conditions is done using the configuration file and the it is expected to either be `.github/dependabot.yml` or `.github/dependabot.yaml`. See the [ruby code in main repo](https://github.com/dependabot/dependabot-core/blob/ef20a02f81275d0ac49232b0a7117eefcf125373/common/lib/dependabot/config/file_fetcher.rb#L9). This particular line of code is relied upon at 
https://github.com/tinglesoftware/dependabot-azure-devops/blob/b179ed6506f860603af2b1790475bbddeecead61/updater/bin/update-script.rb#L458-L468

If you can get that changed then we can merge this. Even then, maybe we end up fixing to `.azuredevops/dependabot.{yaml\yml}` unless you can make the config file fetcher dynamic too.

**Comment by DaleMckeown** [comment]
@mburumaxwell Thanks, I wasn't aware of the upstream requirement. I'll see what I can do.

**Comment by DaleMckeown** [comment]
@mburumaxwell  I've had a go at this! I can't run the project so it's not tested, I may have missed some bits.

**Comment by DaleMckeown** [comment]
@mburumaxwell I think I've reached the limit of my skillset with this, my lack of knowledge in Ruby is hampering my ability to add the unit tests as requested.

Feel free to abandon this if no one can finish polishing it.

**Comment by mburumaxwell** [comment]
Sorry, this is exactly what happens when the community asks for a feature. Asking is the easy part, implementing is always the hard part, especially when one has to add tests too.

I can't promise to get on this soon. Hopefully, someone else in need of the same can get it done.

I am certain that there are workarounds for this.

**Comment by mburumaxwell** [comment]
@DaleMckeown , as per #884 and #885, we no longer parse the dependabot.yml file in the Ruby script but instead pass the necessary details via ENV. This means we can read a file from anywhere. Could you relook into this now? You'd only need to make changes in the extension and server components.

**Pull request closed without merging**